src/Component/PopUp/Popup.js

```diff
@@ -0,0 +1,119 @@
+import React, { useEffect } from "react";
+import { useState } from "react";
+import moment from "moment";
+export const Popup = ({ popupModal, setpopupModal, inputVal, setInputVal, showItems, setShowItems, editIndexField, setEditIndexField }) => {
+    const [error, setError] = useState(false)
+    const [isDoneButtonDisabled, setDoneButtonDisabled] = useState(false);
+    const currentDateAndTime = moment().format("YYYY-MM-DDTHH:mm");
+    const handleSubmit = (e) => {
+        e.preventDefault()
+        if (inputVal.item.trim() === "") {
+            setError("Please enter a text.");
+            setDoneButtonDisabled(true);
+            return;
+        }
+        if (editIndexField >= 0) {
+            let color = "purpleDot"
+            const updatedItems = [...showItems];
+            updatedItems.splice(editIndexField, 1, { item: inputVal.item, dateTime: inputVal.dateTime, taskColor: color });
+            setShowItems(updatedItems);
+            setInputVal({
+                item: "",
+            });
+            setEditIndexField(-1);
+        } else {
+            setShowItems([...showItems, inputVal]);
+            setInputVal({
+                item: "",
+            });
+            setError(false);
+        }
+        setpopupModal(false)
+
+    }
+
+    const handleChange = (e) => {
+        const { name, value } = e.target
+
+        const selectedDateTime = moment(value);
+
+        const currentDateTime = moment();
+
+        if (selectedDateTime.isBefore(currentDateTime)) {
+            setError("Please select a future date and time.");
+            setDoneButtonDisabled(true);
+            return;
+        }
+
+
+        let color = "";
+        const selectedDate = moment(inputVal.dateTime).format("YYYY-MM-DD")
+        const currentDate = moment(new Date()).format("YYYY-MM-DD")
+        if (moment(selectedDate).isSameOrAfter(currentDate, 'day')) {
+            color = "purpleDot";
+        } else {
+            color = "red";
+        }
+        setInputVal((previousData) => {
+            return {
+                ...previousData,
+                [name]: value,
+                isCompleted: false,
+                taskColor: color,
```

Comment:
why do you need color variable, you can directly add moment condition here @amrit56789